### PR TITLE
Port to meson build system

### LINKED ISSUE #309: Migrate to meson

### PR DESCRIPTION
**Pull request by tintou**
fixes #309

### CODE REVIEW

**Comment by tintou** [comment]
this will fail until we add the missing .po files (only need to run `ninja extra-update-p` and `ninja io.elementary.photos-update-po`

**Comment by tintou** [comment]
Note that in this port everything is now using GResources

**Comment by tintou** [comment]
There is an option to enable documentation generation (run `meson configure -Ddocumentation=true`)

File: README.md [hunk 1]
```diff
@@ -32,17 +32,12 @@ You'll need the following dependencies:
...
 
-It's recommended to create a clean build environment
+Run `meson build` to configure the build environment. Change to the build directory and run `ninja test` to build
 
-    mkdir build
-    cd build/
-    
-Run `cmake` to configure the build environment and then `make` to build
+    meson build --prefix=/usr
+    cd build
```
Comment (danrabbit) [inline README.md]: Did you forget to write `ninja test` here? :)

File: po/extra/meson.build [hunk 2]
```diff
@@ -0,0 +1,3 @@
+i18n.gettext('extra',
```
Comment (davidmhewitt) [inline po/extra/meson.build]: We don't want to install these, right?
Should have an `install: false`?

**Comment by tintou** [comment]
So everything should be working by now but travis is failing because of it being based on xenial

**Comment by codygarver** [comment]
I get this startup crash:
```
[FATAL 21:36:18.311876] Plugins.vala:454: Unable to load module /usr/lib/x86_64-linux-gnu/io.elementary.photos/plugins/libpantheon-photos-transitions.so (SPIT 0): no spit module returned
```

If I delete all plugins it launches fine.

I am building from packaging branch `deb-packaging-juno`.

**Comment by tintou** [comment]
@danrabbit @codygarver crash is now fixed

**Pull request merged**